src/main/java/io/dropwizard/revolver/resource/RevolverRequestResource.java

```diff
@@ -203,7 +203,7 @@ private Response processRequest(final String service, final RevolverHttpApiConfi
                 }
                 return executeCommandAsync(service, apiMap.getApi(), method, path, headers, uriInfo, body, apiMap.getApi().isAsync(), callMode);
             case RevolverHttpCommand.CALL_MODE_CALLBACK_SYNC:
-                if(Strings.isNullOrEmpty(headers.getHeaderString(RevolversHttpHeaders.CALLBACK_URI_HEADER)) || apiMap.getApi().isAsync()) {
```

Comment:
@phaneesh Mercury is still calling the async api with callback mode as sync.  :(
They will move to either callback or polling for calling the async api